test/it/controller/etcdopstask/assertions.go

```diff
@@ -0,0 +1,94 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package etcdopstask
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/core/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/gomega"
+)
+
+// assertEtcdOpsTaskStateAndErrorCode waits for the EtcdOpsTask to reach the specified state with the expected error code (or nil for success) in the given phase.
+func assertEtcdOpsTaskStateAndErrorCode(ctx context.Context, g *WithT, t *testing.T, cl client.Client, etcdOpsTask *druidv1alpha1.EtcdOpsTask, expectedTaskState druidv1alpha1.TaskState, expectedOperationState druidv1alpha1.LastOperationState, expectedPhase druidv1alpha1.OperationPhase, expectedErrorCode *druidv1alpha1.ErrorCode) {
```

Comment:
The order of arguments needs to be corrected based on agreed upon hierarchy (phase vs state)